### PR TITLE
Add a flag to build.{sh,cmd} to disable PGO.

### PR DESCRIPTION
**Pull request by pgavlin**
This is helpful when performing JIT throughput measurements.

### CODE REVIEW

**Comment by pgavlin** [comment]
@dpodder @BruceForstall PTAL

**Comment by BruceForstall** [comment]
Related to #12128

File: build.cmd [hunk 1]
```diff
@@ -138,6 +139,7 @@ if /i "%1" == "skipbuildpackages"   (set __BuildPackages=0&set processedArgs=!pr
 if /i "%1" == "skiprestoreoptdata"  (set __RestoreOptData=0&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
 if /i "%1" == "usenmakemakefiles"   (set __NMakeMakefiles=1&set __ConfigureOnly=1&set __BuildNative=1&set __BuildNativeCoreLib=0&set __BuildCoreLib=0&set __BuildTests=0&set __BuildPackages=0&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
 if /i "%1" == "pgoinstrument"       (set __PgoInstrument=1&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
+if /i "%1" == "nopgooptimize"       (set __PgoOptimize=0&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
 if /i "%1" == "ibcinstrument"       (set __IbcTuning=/Tuning&set processedArgs=!processedArgs! %1&shift&goto Arg_Loop)
```
Comment (BruceForstall) [inline build.cmd]: You need to add this to the "usage" message, around line 630

File: build.sh [hunk 2]
```diff
@@ -730,6 +731,11 @@ while :; do
             __PgoInstrument=1
             ;;
 
+        nopgooptimize)
+            __PgoOptimize=0
```
Comment (BruceForstall) [inline build.sh]: Add to the usage message, around line 40

File: build.cmd (hunk 1 above)
Comment (pgavlin) [inline build.cmd]: Done.

File: build.sh (hunk 2 above)
Comment (pgavlin) [inline build.sh]: Done.

File: build.cmd [hunk 3]
```diff
@@ -158,6 +160,8 @@ if [!processedArgs!]==[] (
 
 :ArgsDone
 
+if %__PgoOptimize%=0 set __RestoreOptData=0
```
Comment (dpodder) [inline build.cmd]: Wrong syntax--you need double-equals. To keep with script conventions, I recommend either
```batch
if "%__PgoOptimize%" == "0" set __RestoreOptData=0
```
or
```batch
if [%__PgoOptimize%] == [0] set __RestoreOptData=0
```

File: build.cmd [hunk 4]
```diff
@@ -620,6 +624,7 @@ echo     for the specified platform ^(FreeBSD, Linux, NetBSD, OS X or Windows,
 echo     respectively^).
 echo     add nativemscorlib to go further and build the native image for designated mscorlib.
 echo toolset_dir ^<dir^> : set the toolset directory -- Arm64 use only. Required for Arm64 builds.
+echo nopgooptimize - do not use profile guided optimizations.
```
Comment (BruceForstall) [inline build.cmd]: super-nit: the cmd file standard seems to be ": ", not " - " (like in the build.sh file).

**Comment by pgavlin** [comment]
> I had previously discussed with @janvorli to refactor this to be a single mode to support multiple states, e.g.: `pgo:instrument`, `pgo:force`, `pgo:auto` (default), or `pgo:off`. Could we go with the latter name for the new switch, so that we don't have to introduce a breaking change later? Or is `nopgooptimize` preferred, and should we add `forcepgooptimize` later?

I'm not too concerned with making a breaking change later. This change is meant to ease some of the throughput investigations the JIT team is currently performing, so I'd prefer to leave it as-is and get it checked in sooner.

File: build.cmd (hunk 4 above)
Comment (pgavlin) [inline build.cmd]: Fixed.

**Comment by dpodder** [comment]
Sounds good. LGTM.

**Comment by pgavlin** [comment]
@dotnet-bot Tizen armel Cross Debug Build

**Comment by pgavlin** [comment]
@dotnet-bot test Tizen armel Cross Debug Build

**Pull request merged**